[Macro] Crash on malformed case statement with multiple else

The following macro generates two `else` for the case statement and fails when compiling C. I'd expect Nim to throw a compilation error. The error is less obvious when constructing the case dynamically.

### Example
```nim
import macros

macro genCase(): untyped =
  result = newStmtList()
  result.add(
    nnkStmtList.newTree(
    nnkVarSection.newTree(
      nnkIdentDefs.newTree(
        newIdentNode("foo"),
        newEmptyNode(),
        newLit(1)
      )
    ),
    nnkCaseStmt.newTree(
      newIdentNode("foo"),
      nnkOfBranch.newTree(
        newLit(1),
        nnkStmtList.newTree(
          nnkCommand.newTree(
            newIdentNode("echo"),
            newLit(1)
          )
        )
      ),
      nnkElse.newTree(
        nnkStmtList.newTree(
          nnkCommand.newTree(
            newIdentNode("echo"),
            newLit(0)
          )
        )
      ),
      nnkElse.newTree(
        nnkStmtList.newTree(
          nnkCommand.newTree(
            newIdentNode("echo"),
            newLit(0)
          )
        )
      )
    )
  )
  )
  echo repr(result)
  #[
  var foo = 1
  case foo
  of 1:
    echo 1
  else:
    echo 0
  else:
    echo 0
  ]#

proc main() =
  genCase()
main()
```

### Current Output
```
Error: execution of an external compiler program 'gcc -c  -w   -I'/xxx/.choosenim/toolchains/nim-#devel/lib' -I/xxx/crap -o /xxx/.cache/nim/somemacro_d/@msomemacro.nim.c.o /xxx/.cache/nim/somemacro_d/@msomemacro.nim.c' failed with exit code: 1

/xxx/.cache/nim/somemacro_d/@msomemacro.nim.c: In function ‘main__an59b5mnLzeWCeQGX37ps2Q’:
/xxx/.cache/nim/somemacro_d/@msomemacro.nim.c:157:2: error: multiple default labels in one switch
  default:
  ^~~~~~~
/xxx/.cache/nim/somemacro_d/@msomemacro.nim.c:152:2: note: this is the first default label
  default:
  ^~~~~~~
```

### Expected Output

A Nim compilation error.

### Additional Information

* I tried in Nim devel and 1.0.4

```
$ nim -v
Nim Compiler Version 1.1.1 [Linux: amd64]
Compiled at 2020-01-24
Copyright (c) 2006-2019 by Andreas Rumpf

active boot switches: -d:release
```

--- a/compiler/semstmts.nim
+++ b/compiler/semstmts.nim
@@ -945,9 +945,9 @@ proc semCase(c: PContext, n: PNode; flags: TExprFlags): PNode =
       checkSonsLen(x, 1, c.config)
       x[0] = semExprBranchScope(c, x[0])
       typ = commonType(typ, x[0])
-      hasElse = true
-      if chckCovered and covered == toCover(c, n[0].typ):
+      if (chckCovered and covered == toCover(c, n[0].typ)) or hasElse:
         localError(c.config, x.info, "invalid else, all cases are already covered")
+      hasElse = true
       chckCovered = false
     else:
       illFormedAst(x, c.config)

